hy/compiler.py

```diff
@@ -503,19 +503,13 @@ def _parse_lambda_list(self, exprs):
         for expr in exprs:
 
             if expr in ll_keywords:
-                if expr == "&rest" and lambda_keyword is None:
-                    lambda_keyword = expr
-                elif expr == "&optional":
+                if expr == "&optional":
                     if len(defaults) > 0:
                         raise HyTypeError(expr,
                                           "There can only be &optional "
                                           "arguments or one &key argument")
                     lambda_keyword = expr
```

Comment:
Now I wonder if this whole thing can be reduced to:

```python
if expr in ("&optional", "&rest", "&key", "&kwonly", "&kwargs"):
    if len(defaults) > 0:
        # ...
    lambda_keyword = expr
```

(This isn't necessary for this PR; it's just me wondering out load.)